app/src/main/java/com/monkeyteam/chimpagne/ui/DetailScreenSheet.kt

```diff
@@ -1,94 +1,67 @@
-import androidx.compose.runtime.setValue
-import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
-import androidx.compose.ui.platform.testTag
-import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.input.nestedscroll.NestedScrollConnection
+import androidx.compose.ui.input.nestedscroll.nestedScroll
+import androidx.compose.ui.platform.LocalConfiguration
+import androidx.compose.ui.unit.Velocity
 import androidx.compose.ui.unit.dp
-import com.monkeyteam.chimpagne.R
 import com.monkeyteam.chimpagne.model.database.ChimpagneEvent
-import com.monkeyteam.chimpagne.model.utils.createCalendarIntent
-import com.monkeyteam.chimpagne.ui.components.SimpleTagChip
-import com.monkeyteam.chimpagne.ui.components.popUpCalendar
-import com.monkeyteam.chimpagne.ui.theme.ChimpagneTypography
+import com.monkeyteam.chimpagne.ui.components.EventCard
 
 @Composable
-fun DetailScreenSheet(
-    event: ChimpagneEvent?,
-    onJoinClick: () -> Unit = {},
-    context: Context? = null
+fun DetailScreenListSheet(
+    events: List<ChimpagneEvent>,
+    onEventClick: (ChimpagneEvent) -> Unit = {}
 ) {
-  var showDialog by remember { mutableStateOf(false) }
-  val enhancedOnJoinClick = {
-    onJoinClick()
-    showDialog = true
-  }
-  if (event != null && event.id.isNotBlank()) {
-    Column(
-        modifier = Modifier.fillMaxWidth().padding(16.dp),
-        horizontalAlignment = Alignment.CenterHorizontally) {
-          Text(
-              text = event.title,
-              style = ChimpagneTypography.headlineMedium,
-              modifier = Modifier.padding(bottom = 8.dp))
-
-          Text(
-              text = event.startsAt().time.toString(),
-              style = ChimpagneTypography.bodyMedium,
-              modifier = Modifier.padding(bottom = 8.dp))
-
-          Text(
-              text = event.endsAt().time.toString(),
-              style = ChimpagneTypography.bodyMedium,
-              modifier = Modifier.padding(bottom = 8.dp))
+  val screenHeight = LocalConfiguration.current.screenHeightDp.dp
+  val maxSheetHeight = screenHeight * 0.75f
 
-          Text(
-              text = event.description,
-              style = ChimpagneTypography.bodySmall,
-              modifier = Modifier.padding(bottom = 8.dp))
+  // State to keep track of the LazyColumn scroll position
+  val lazyListState = rememberLazyListState()
+  val nestedScrollConnection = remember {
+    object : NestedScrollConnection {
 
-          Row(
-              modifier = Modifier.fillMaxWidth().padding(bottom = 8.dp),
-              horizontalArrangement = Arrangement.SpaceEvenly) {
-                event.tags.forEach { tag -> SimpleTagChip(tag) }
-              }
+      override suspend fun onPostFling(consumed: Velocity, available: Velocity): Velocity {
+        val isAtTop =
+            lazyListState.firstVisibleItemIndex == 0 &&
+                lazyListState.firstVisibleItemScrollOffset == 0
+        val isAtBottom =
```

Comment:
This is hard to read. Consider breaking down the expression into digestible pieces